sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferClient.java

```diff
@@ -52,43 +56,48 @@
      * for transfers.
      *
      * @param context the context
-     * @param blobClient the blob storage client
      * @param constraints the constraints to meet to run transfers
      * @param serialTaskExecutor the executor for all internal book keeping purposes
+     * @param storageBlobClients the blob storage clients for transfers
      */
     private TransferClient(Context context,
-                           StorageBlobClient blobClient,
                            Constraints constraints,
-                           SerialExecutor serialTaskExecutor) {
+                           SerialExecutor serialTaskExecutor,
+                           Map<String, StorageBlobClient> storageBlobClients) {
         this.context = context;
-        this.blobClient = blobClient;
         this.constraints = constraints;
         this.serialTaskExecutor = serialTaskExecutor;
         this.db = TransferDatabase.get(context);
+        STORAGE_BLOB_CLIENTS.putAll(storageBlobClients);
     }
 
     /**
      * Upload a file.
      *
+     * @param storageBlobClientId the identifier of the blob storage client to use for the upload
      * @param containerName the container to upload the file to
      * @param blobName the name of the target blob holding uploaded file
      * @param file the local file to upload
      * @return LiveData that streams {@link TransferInfo} describing current state of the transfer
      */
-    public LiveData<TransferInfo> upload(String containerName, String blobName, File file) {
+    public LiveData<TransferInfo> upload(String storageBlobClientId, String containerName, String blobName, File file) {
```

Comment:
What does the public API look like for this? That is, can you give me a snippet to show how a user would call this upload method? I'm trying to recall why we would have the client ID passed here rather than being able to retrieve it from the client instance.

Reply:
The usage as of today will look like:

```java
// Create raw client
StorageBlobClient storageBlobClient = new StorageBlobClient.Builder()
                .setBlobServiceUrl(baseUrl)
                .setCredentialInterceptor(getSaSAuthInterceptor())
                .build();

// create transfer client
TransferClient transferClient = new TransferClient.Builder(getApplicationContext())
            .addStorageBlobClient("myapp.photo", storageBlobClient)
            .setRequiredNetworkType(androidx.work.NetworkType.CONNECTED)
            .build();

transferClient.upload(("myapp.photo", "mypic.jpg", file)
    .observeForever(transferInfo -> {

     });
```

Reply:
Design is a bit different from iOS where TransferManager is internal and the user interacts with StorageBlobClient.

I like to explore exposing APIs in StorageBlobClient and hide TransferClient, still, TransferClient has to take id since it allows registering multiple StorageBlobClient with it, which is the same case of iOS TransferManager I believe.